addons/account_peppol/models/res_company.py

```diff
@@ -32,6 +66,72 @@ class ResCompany(models.Model):
         compute='_compute_peppol_purchase_journal_id', store=True, readonly=False,
     )
 
+    # -------------------------------------------------------------------------
+    # HELPER METHODS
+    # -------------------------------------------------------------------------
+
+    def _sanitize_peppol_phone_number(self):
+        self.ensure_one()
+
+        error_message = _(
+            "Please enter the phone number in the correct international format.\n"
+            "For example: +32123456789, where +32 is the country code.\n"
+            "Currently, only European countries are supported.")
+
+        if not phonenumbers:
+            raise ValidationError(error_message)
+
+        phone_number = self.account_peppol_phone_number
+        if not phone_number.startswith('+'):
+            phone_number = f'+{phone_number}'
+
+        try:
+            phone_nbr = phonenumbers.parse(phone_number)
+        except phonenumbers.phonenumberutil.NumberParseException:
+            raise ValidationError(error_message)
+
+        country_code = phonenumbers.phonenumberutil.region_code_for_number(phone_nbr)
+        if country_code not in PHONE_ALLOWED_COUNTRIES or not phonenumbers.is_valid_number(phone_nbr):
+            raise ValidationError(error_message)
+
+    def _check_peppol_endpoint_number(self, peppol_dict):
+        self.ensure_one()
+
+        endpoint_rule = peppol_dict.get(self.peppol_eas)
+        if not endpoint_rule:
+            return False
+
+        if endpoint_rule[0] == 'regex':
+            return bool(re.match(endpoint_rule[1], self.peppol_endpoint))
+
+        if endpoint_rule[0] == 'ean':
+            check_module = ean
+        else:
+            check_module = get_cc_module(endpoint_rule[0], endpoint_rule[1])
+        return check_module.is_valid(self.peppol_endpoint)
+
+    # -------------------------------------------------------------------------
+    # CONSTRAINTS
+    # -------------------------------------------------------------------------
+
+    @api.constrains('account_peppol_phone_number')
+    def _check_account_peppol_phone_number(self):
+        for company in self:
+            if company.account_peppol_phone_number:
+                company._sanitize_peppol_phone_number()
+
+    @api.constrains('peppol_endpoint')
+    def _check_peppol_endpoint(self):
+        for company in self:
+            if not company.peppol_endpoint or company.peppol_eas not in PEPPOL_ENDPOINT_RULES:
+                continue
+            if not company._check_peppol_endpoint_number(PEPPOL_ENDPOINT_RULES):
```

Comment:
put `PEPPOL_ENDPOINT_RULES` and `PEPPOL_ENDPOINT_RULES` in company and check if the endpoint/eas is set directly on the method ?